api: Call Sendgrid email validation API on signup

victorges opened the pull request:
<!-------------------------------------------------------------------------
 | Thanks for send a pull request! 🎉
 | First, please make sure you familiar with the contribution guidelines
 | https://github.com/livepeer/livepeer.com/blob/master/CONTRIBUTING.md
 -------------------------------------------------------------------------->

**What does this pull request do? Explain your changes. (required)**

This is to test Sendgrid email validation API to see if it would be interesting for us.
Runs totally in background and only logs the result, so not to pollute our DB with any
data nor affect the API performance. With loki we should be able to do some smart 
queries to know how well it is performing.

Let's keep this testing new signups and check how accurate it would be to detect
malicious users like soccer streamers.

The idea is that we would eventually **require** email verification only when the repsonse
from this API is something like `Risky` or `Invalid`.

**Specific updates (required)**
Add a function that calls sengrid verification API and logs the result.

## -

- **How did you test each of these updates (required)**
Will test on staging where we deploy the API with a sendgrid key. 
Might need to change the key to have access to email verification API though.

**Does this pull request close any open issues?**
Not really. But works on improving the email verification flow which now is totally open.

**Checklist:**

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have read the **CONTRIBUTING** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

victorges committed:
api: Create api helper to validate an email address

victorges committed:
api: Call validate email func on signup

hjpotter92 commented on packages/api/src/controllers/helpers.js:
```diff
@@ -163,6 +164,32 @@ export async function sendgridEmail({
+    method: "POST",
+    body: { email, source: "signup" },
+  };
+  client
```
typo here? should it be `SendgridClient`?

victorges replied:
Ugh yes, thanks for spotting that. I hate JS, doesn't warn even on a typo 🙄

victorges replied:
Couldn't hold myself. Converted these helpers to TypeScript 😡

hjpotter92 replied:
:rofl: I was going to suggest maybe make it a module? `helpers/sendgrid.ts` etc?

victorges replied:
I think we can do that if we end up keeping this email validation logic. Otherwise, it might be just 1 function in there for sending emails.

victorges committed:
api: Fix typo

victorges committed:
api: Convert helpers to TypeScript

victorges committed:
api: Separate async email validation function

Cleaner than the .then.catch IMO

victorges committed:
api: Receive separate arg for validating email addresses

I wouldn't do that if Sendgrid didn't obly us to do that :sweat:

They don't support API keys that can both send emails and validate
addresses. I know wtf.

victorges committed:
api: Add back isomorphic-webcrypto

victorges committed:
api/helpers: Remove unused imports

victorges committed:
api/users: Add FieldsMap type

victorges committed:
api: Fix tests

In the end it was a problem with our mock which wasn't
following the interface. Decided to just revert the old
change to be compatible instead regarding the personalization.to
but changed the snakes to camels instead.

victorges committed:
api: Do not call validation API if no API key provided
